### ISSUE TITLE
Quotation marks

### ISSUE THREAD

**Issue by dextermb** [langleyfoxall/technologies#68]
### What you would like to change/add
I would like to propose that single quote marks should be the company standard.

### Why you would like to change/add this
There's a lot of inconsistencies at the moment, it would help make the code look tidier if we all used the same quote marks.

### Examples
Quotation marks are probably one of the most used symbols when programming.

#### Basic usage
```php
$var = 'Hello, world';
```

#### Need to have a single quote within the string? Escape it
```php
$var = 'Anthony \'Ant\' Fowler';
```

#### Need to append variables? Use `.`
_Note that any string operation is slow with PHP, but appending strings is quicker than evaluating variables within a string._

```php
$var = 'Hello, ' . $world;
```

### Checklist
- [x] I have checked that this is not a duplicate issue.

**Comment by ash123456789** [comment]
I agree with this for the most part apart from the portion on appending variables.

If you need to append multiple variables to a string, putting them inside the quotes makes thing a lot easier to read. Yes, it is slower - but the difference is extremely minimal.

**Comment by wheatleyjj** [comment]
I like using double when doing interpolation.

**Comment by DivineOmega** [comment]
@dextermb Strongly agree on all points.

**Comment by DivineOmega** [comment]
@ash123456789 You can not include variables in single quotes strings.

> Note: Unlike the double-quoted and heredoc syntaxes, variables and escape sequences for special characters will not be expanded when they occur in single quoted strings.

https://secure.php.net/manual/en/language.types.string.php#language.types.string.syntax.single

**Comment by dextermb** [comment]
@ash123456789 you could also just do
```
"Hello, $world"
```

**Comment by AlexCatch** [comment]
These change based on the context.

Single quotes should be the standard for 99% of use cases but I know in React passing a string as a prop you use double quotation marks.

As long as we deal with that, single quotation marks are the way to go.

**Comment by DivineOmega** [comment]
@AlexCatch Good point Alex.

@dextermb Are we talking about PHP primarily or all languages? Might be worth clarifying in the issue title.

**Comment by dextermb** [comment]
Although I am a fan of using backticks in JavaScript and using variables within the string. For example:
```
`Hello, ${world}`
```

Which I think should be another exemption.

**Comment by JamesLF** [comment]
One problem with single quote is that the apostrophe is far more common in written language than quote marks. It's likely to far more often be used in our application's interface text and so we're likely to have to escape it much more often than speech-marks.
There's four in this post alone!

**Comment by AlexCatch** [comment]
@dextermb While I'd often agree with concatenation over interpolation, in JS I think it's a lot cleaner than concatenation & many style guides often prefer you to use interpolation over concatenation.

In terms on speed, they seem to be mostly the same although it's hard to get a solid consensus.

**Comment by dextermb** [comment]
@AlexCatch On another note, for React you use `=""` for props rather than `={''}`?

Examples:

Double quotes:
```
<div className="cool-class" />
```

Brace and single quotes:
```
<div className={'cool-class'} />
```

**Comment by AlexCatch** [comment]
@dextermb The AirBnb React style guide specifies not using brackets for string props as it's unnecessary.

**Comment by dextermb** [comment]
> @dextermb The AirBnb React style guide specifies not using brackets for string props as it's unnecessary.

Ah, I see. Alright 👍

**Comment by dextermb** [comment]
@AlexCatch At some point we will have to get a copy of the [Airbnb styleguide for React](https://github.com/airbnb/javascript/tree/master/react) so that we can start enforcing stuff.
